BSP/STM32F413/Src/BSP_UART.c

```diff
@@ -166,25 +166,28 @@ uint32_t BSP_UART_Read(UART_t usart, char *str) {
     uint32_t recvd = 0;
 
     bool *line_recvd = lineRecvd[usart];
-    if(*line_recvd) {
-        USART_TypeDef *usart_handle = handles[usart];
+    
+    while(*line_recvd == false){
+        BSP_UART_Write(usart, "", 0);   // needs to be in. Otherwise, usbLineRecieved will not update
+    }
 
-        USART_ITConfig(usart_handle, USART_IT_RXNE, RESET);
-        
-        rxfifo_t *fifo = rx_fifos[usart];
-        
-        rxfifo_peek(fifo, &data);
-        while(!rxfifo_is_empty(fifo) && data != '\r') {
-            recvd += rxfifo_get(fifo, (char*)str++);
-            rxfifo_peek(fifo, &data);
-        }
+    USART_TypeDef *usart_handle = handles[usart];
 
-        rxfifo_get(fifo, &data);
-        *str = 0;
-        *line_recvd = false;
-        USART_ITConfig(usart_handle, USART_IT_RXNE, SET);
+    USART_ITConfig(usart_handle, USART_IT_RXNE, RESET);
```

Comment:
just test uart read to make sure it works, because afaik, we haven't actually used it anywhere important before

Reply:
We've tested and it does work.